R/moma_arguments.R

```diff
@@ -389,21 +264,226 @@ create_moma_sparsity_func <- function(f) {
+#' are provided. When "path" is chosen, the algorithm by
+#' Hoefling, H. (2010) is used. When "dp" is chosen, the algorithm by Johnson, N. A. (2013) is used.
+#'
+#' @references Tibshirani, Robert, et al. "Sparsity and Smoothness via the Fused Lasso."
+#' Journal of the Royal Statistical Society: Series B (Statistical Methodology) 67.1 (2005): 91-108.
+#' \doi{10.1111/j.1467-9868.2005.00490.x}.
+#' @references Hoefling, H. (2010). A path algorithm
+#' for the fused lasso signal approximator. Journal of Computational and Graphical
+#'  Statistics, 19(4), 984-1006. \doi{10.1198/jcgs.2010.09208}.
+#' @references Johnson, N. A. (2013). A dynamic programming algorithm for the
+#' fused lasso and l 0-segmentation. Journal of Computational and Graphical
+#' Statistics, 22(2), 246-260. \doi{10.1080/10618600.2012.681238}.
+#' @return A \code{moma_sparsity_type} object, which is an empty list.
+#'
+#' @name fused_lasso
+#' @inheritParams moma_sparsity_options
+#' @export
 moma_fusedlasso <- create_moma_sparsity_func(fusedlasso)
+
+#' L1 trend filtering
+#'
+#' Use this function to set the penalty function to l1 trend filtering. An
+#' important special case is when \eqn{k=1}. In this case the penalty
+#' term becomes
+#' \deqn{\lambda \sum | x_{i-1} - 2x_{i} + x_{i+1} |,}
+#' where \eqn{\lambda} is set by the \code{lambda} argument below.
+#'
+#' The general formula of the penalty term for \eqn{k \in N} can be found in
+#' the paper cited in Reference. For other values of \eqn{k} please refer to the following table:
+#' \tabular{llll}{
+#'   \tab \eqn{k = 0} \tab \eqn{k = 1} \tab \eqn{k = 2}
+#' \cr
+#'  Type of sparsity \tab piecewise constant \tab peicewise linear \tab piecewise quadratic
+#' }
+#'
+#' @param ... Forces users to specify all arguments by name.
+#' @param l1tf_k Use (\eqn{k+1})-difference matrix in trend filtering. Note \eqn{k = 0}
+#'          implies piecewise constant, \eqn{k=1} implies piecewise linear, \eqn{k=2}
+#'          piecewise quadratic etc.
+#' @references Tibshirani, Ryan J. "Adaptive Piecewise Polynomial Estimation via Trend
+#' Filtering." The Annals of Statistics 42.1 (2014): 285-323. \doi{10.1214/13-AOS1189}.
+#' @return A \code{moma_sparsity_type} object, which is an empty list.
+#' @name l1_trend_filtering
+#' @inheritParams moma_sparsity_options
+#' @export
 moma_l1tf <- create_moma_sparsity_func(l1tf)
+
+#' Sparse fused LASSO
+#'
+#' Use this function to set the penalty function to sparse fused lasso
+#' \deqn{\lambda_1 \sum | x_{i} - x_{i-1} | + \lambda_2 \sum |x_{i} | ,}
+#' where \eqn{\lambda_1} is set by the \code{lambda} argument below, and \eqn{\lambda_2}
+#' is specified in by the \code{lambda_2} argument.
+#'
+#' @param ... Forces users to specify all arguments by name.
+#' @param lambda2 A scalar. The level of penalty on the absolute values of the coefficients.
+#'
+#' @return A \code{moma_sparsity_type} object, which is a list containing the value of \code{lambda_2}.
```

Comment:
Maybe make a bit clearer that lambda_2 remains fixed, even as we allow searching over lambda_1 in the usual way, so folks don't expect glmnet like behavior (i.e., where the lambda_1 / lambda_2 ratio remains fixed).

Reply:
Fixed.